### PR TITLE
viewing data from old months

### PR DESCRIPTION
**Pull request by gabrielkay**
- Adds `previous month` and `next month` buttons under graphs to navigate to other months data
- `next month` only shows up when viewing past months
- adds CSS styling to the goals show page and index page

Screen Shot 2019-07-29 at 11 16 09 AM

Screen Shot 2019-07-29 at 10 04 45 AM

Screen Shot 2019-07-29 at 10 04 57 AM

### CODE REVIEW

File: app/views/goals/index.html.erb [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+    <div class="goals">
+      <div class="card border-dark mb-3" style="max-width: 18rem;">
+        <% log_sum = Goal.month_progress_sum(goal.id, 0) %>
+        <% goal_sum = goal.frequency %>
+        <% percent = (100 * log_sum / goal_sum).ceil %>
+        <% percent = 100 if percent > 100 %>
+        <div class="card-header"><%= link_to goal.name, goal_path(goal.id) %></div>
+          <div class="card-body text-dark">
+          <p class="card-text"><%= "#{log_sum}/#{goal_sum} #{goal.unit}" %></p>
+        </div>
```
Comment (albaer) [inline app/views/goals/index.html.erb]: This is a lot of math/logic for a view. Moving some of this to the controller or model would be good.

File: app/controllers/goals_controller.rb [hunk 2]
```diff
@@ -9,8 +9,10 @@ def index
   def show
     @goal = Goal.find(params[:id])
     @log = Log.new
-    @progress = Goal.month_progress_sum(@goal.id)
-    @month = Date.today.strftime("%B")
+    @months_ago = (params[:months_ago] || 0).to_i
```
[the remark this answers is not in the document]
Comment (gabrielkay) [reply app/controllers/goals_controller.rb]: It's used a lot in the show.html.erb. That value gets passed to the graphql query and is used for the links to previous/next month.

File: app/views/goals/index.html.erb [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
-    <% log_sum = Goal.month_progress_sum(goal.id) %>
-    <% goal_sum = goal.frequency %>
-    <% percent = (100 * log_sum / goal_sum).ceil %>
-    <% percent = 100 if percent > 100 %>
-    <div class="card-header"><%= link_to goal.name, goal_path(goal.id) %></div>
-      <div class="card-body text-dark">
-      <p class="card-text"><%= "#{log_sum}/#{goal_sum} #{goal.unit}" %></p>
+<div class="index">
+  <% @goals.each do |goal| %>
+    <div class="goals">
```
[the remark this answers is not in the document]
Comment (gabrielkay) [reply app/views/goals/index.html.erb]: meant to be inside, but definitely should have named it singular `goal`

File: app/models/goal.rb [hunk 4]
```diff
@@ -13,7 +13,7 @@ def self.month_progress(goal_id, months_ago)
     Goal.find(goal_id).logs.where(completed_on: beginning_of_month..beginning_of_next_month)
   end
 
-  def self.month_progress_sum(goal_id)
-    Goal.month_progress(goal_id, 0).sum('quantity')
+  def self.month_progress_sum(goal_id, months_ago)
+    Goal.month_progress(goal_id, months_ago).sum('quantity')
```
[the remark this answers is not in the document]
Comment (gabrielkay) [reply app/models/goal.rb]: 👀  I changed them all to instance methods

File: app/views/goals/index.html.erb (hunk 1 above)
Comment (gabrielkay) [reply app/views/goals/index.html.erb]: I created two new methods in the goal model to move this logic out of the view.

File: app/controllers/goals_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (albaer) [reply app/controllers/goals_controller.rb]: Carry on, then.

File: app/models/goal.rb [hunk 5]
```diff
@@ -7,13 +7,26 @@ class Goal < ApplicationRecord
...
+    month_progress(months_ago).sum('quantity')
+  end
+
+  def month_progress_string(months_ago)
+    sum = month_progress_sum(months_ago)
+    "#{sum}/#{frequency} #{unit}"
+  end
+
+  def month_progress_percent(months_ago)
+    percent = 100 * month_progress_sum(0) / frequency
```
Comment (albaer) [inline app/models/goal.rb]: Can frequency be 0? Just making sure this can't throw.

Comment (gabrielkay) [reply app/models/goal.rb]: can't be 0, it validates > 0 on creation

File: app/models/goal.rb [hunk 6]
```diff
@@ -7,13 +7,26 @@ class Goal < ApplicationRecord
...
+  def month_progress_sum(months_ago)
+    month_progress(months_ago).sum('quantity')
+  end
+
+  def month_progress_string(months_ago)
+    sum = month_progress_sum(months_ago)
+    "#{sum}/#{frequency} #{unit}"
+  end
+
+  def month_progress_percent(months_ago)
```
Comment (albaer) [inline app/models/goal.rb]: This argument is unused, so it should be removed.

File: app/models/goal.rb [hunk 7]
```diff
@@ -7,13 +7,26 @@ class Goal < ApplicationRecord
...
+    percent = 100 * month_progress_sum(0) / frequency
+    if percent > 100
+      percent = 100
+    end
+    percent
```
Comment (albaer) [inline app/models/goal.rb]: No need to change this if you don't want. An alternate approach here would be to take the minimum of the percentage and 100 like this:
```
percent = 100 * month_progress_sum(0) / frequency
[percent, 100].min
```

**Pull request merged**